src/com/civpvp/attrhider/AttrHider.java

```diff
@@ -33,129 +26,84 @@
+					ItemStack old = event.getPacket().getItemModifier().read(0);
+					if(old == null) return;
+					ItemStack clean = new ItemStack(old.getType());
+					if(old.hasItemMeta() && clean.hasItemMeta()) {
+						ItemMeta oldMeta = old.getItemMeta();
+						ItemMeta newMeta = clean.getItemMeta();
+						if(oldMeta instanceof LeatherArmorMeta) {
+							Color color = ((LeatherArmorMeta)oldMeta).getColor();
+							((LeatherArmorMeta)newMeta).setColor(color);
+						}
+						if(oldMeta.hasEnchants()) {
+							clean.addEnchantment(Enchantment.DURABILITY, 1);
+						}
+						if(oldMeta instanceof BannerMeta) {
+							DyeColor base = ((BannerMeta)oldMeta).getBaseColor();
+							List<Pattern> pattern = ((BannerMeta)oldMeta).getPatterns();
+							((BannerMeta)newMeta).setBaseColor(base);
+							((BannerMeta)newMeta).setPatterns(pattern);
+						}
+						clean.setItemMeta(newMeta);
+					}
+					event.getPacket().getItemModifier().write(0, clean);
+				}
+			});
+		
+		ProtocolLibrary.getProtocolManager().addPacketListener(
+			new PacketAdapter(this, ListenerPriority.NORMAL, PacketType.Play.Server.ENTITY_EFFECT) {
+				@Override
+				public void onPacketSending(PacketEvent event) {
+					if(event.getPacket().getIntegers().read(0) != event.getPlayer().getEntityId()) {
+						event.getPacket().getBytes().write(0, (byte) 26);
+						event.getPacket().getBytes().write(1, (byte) 420);
+						event.getPacket().getBytes().write(2, (byte) 0);
+						event.getPacket().getIntegers().write(1, 420);
+					}
+				}
+			});
+		
 		ProtocolLibrary.getProtocolManager().addPacketListener(
-		new PacketAdapter(this, ListenerPriority.NORMAL, new PacketType[] { PacketType.Play.Server.ENTITY_METADATA }) {
-			public void onPacketSending(PacketEvent event) {
-				try {
-					Player observer = event.getPlayer();
-					//Get the entity from the packet
-					Entity entity = (Entity)event.getPacket().getEntityModifier(observer.getWorld()).read(0);
-					//If the entity is not the observer, and the entity is alive, and the entity is not a dragon or wither,
-					//and the entity is not the observer's mount
-					if ((entity != null) && (observer != entity) && ((entity instanceof LivingEntity)) &&
-					  (!(entity instanceof EnderDragon) && !(entity instanceof Wither)) && (entity.getPassenger() != observer)) {
-						event.setPacket(event.getPacket().deepClone());
-						StructureModifier<List<WrappedWatchableObject>> watcher = event.getPacket().getWatchableCollectionModifier();
-					  	for (WrappedWatchableObject watch : watcher.read(0)) {
-							if ((watch.getIndex() == 6) && (((Float)watch.getValue()).floatValue() > 0.0F)) {
-								watch.setValue(20f);
+			new PacketAdapter(this, ListenerPriority.NORMAL, PacketType.Play.Server.ENTITY_METADATA) {
+				@Override
+				public void onPacketSending(PacketEvent event) {
+					Player player = event.getPlayer();
+					Entity entity = getEntityById(event.getPlayer().getWorld(), event.getPacket().getIntegers().read(0));
```

Comment:
This will iterate over a few thousand entities every time the health of any entity changes. Why not just do it the way it was done previously?